pageserver/src/layered_repository.rs

```diff
@@ -1088,10 +1116,11 @@ impl LayeredTimeline {
     ///
     /// NOTE: This has nothing to do with checkpoint in PostgreSQL. We don't
     /// know anything about them here in the repository.
-    fn checkpoint_internal(&self) -> Result<()> {
+    fn checkpoint_internal(&self, force: bool) -> Result<()> {
+        // FIXME: these should be fetched atomically.
         let last_valid_lsn = self.last_valid_lsn.load();
         let last_record_lsn = self.last_record_lsn.load();
-        let prev_record_lsn = self.prev_record_lsn.load();
+        let prev_record_lsn = self.last_record_lsn.load();
```

Comment:
Can we get rid of 1 of these variables, or only do 1 atomic load

Reply:
Ah, that was actually wrong, 'prev_record_lsn' should still be loaded from 'self.prev_record_lsn'. Fixed!

(see https://github.com/zenithdb/zenith/issues/446, though)